ClientCore/ProgramConstants.cs

```diff
@@ -41,7 +54,8 @@ public static class ProgramConstants
         public const int GAME_ID_MAX_LENGTH = 4;
 
         public static readonly Encoding LAN_ENCODING = Encoding.UTF8;
-
+        public static readonly bool ISMONO = isMono ??= Type.GetType("Mono.Runtime") != null;
+        private static readonly bool? isMono;
```

Comment:
> been looking at this for quite some time and still don't get what it means...

We use the ISMONO property to check if we are running under Mono. We can determine this by checking if the type Mono.Runtime exists in the current runtime. This uses reflection however which is an expensive operation.

When getting the value for the ISMONO property:
- If the isMono field is not null we return that cached value.
- If the isMono field is null we assign it the result of `Type.GetType("Mono.Runtime") != null` and return this value.

Reply:
Defining `ISMONO` as a readonly variable instead of a property means the value of `ISMONO` is determined at the beginning, where the value of `isMono` is always false. Try defining `ISMONO` as a readonly *property* so that the value can be determined by `isMono` no matter at what time `isMono` is changed.

Reply:
> Defining `ISMONO` as a readonly variable instead of a property means the value of `ISMONO` is determined at the beginning, where the value of `isMono` is always null. Try defining `ISMONO` as a readonly _property_ so that the value can be determined by `isMono` no matter at what time `isMono` is changed.

This was supposed to be a property indeed, good catch.